consensus/src/pos/error.rs

```diff
@@ -102,3 +108,14 @@ pub enum ConsensusPoSError {
     #[error("Failed to calculate capped balance")]
     FailedToCalculateCappedBalance,
 }
+
+// TODO: include the original chainstate::ChainstateError in each error below.
+#[derive(Error, Debug, PartialEq, Eq, Clone)]
+pub enum ChainstateError {
+    #[error("Failed to obtain epoch for block height {0}")]
+    FailedToObtainEpochData(BlockHeight, Box<dyn DynError>),
+    #[error("Failed to read data of pool {0}")]
+    StakePoolDataReadError(PoolId, Box<dyn DynError>),
+    #[error("Failed to read balance of pool {0}")]
+    PoolBalanceReadError(PoolId, Box<dyn DynError>),
+}
```

Comment:
Why do we need this, with dyn Errors in it?

Reply:
I've replaced `Box<dyn DynError>` with `String` as discussed.